src/main/webapp/js/common/react/components/AuthButton.js

```diff
@@ -0,0 +1,83 @@
+import checkTesting from '../../../checkTesting.js';
+import standardizeFetchErrors from '../../../fetch_handler.js';
+checkTesting();
+
+/**
+ * Login/Logout button that fetches user authorization data to automatically
+ * update its appearance.
+ * @extends React.Component
+ */
+export default class AuthButton extends React.Component {
+  /**
+   * Create the AuthButton react component.
+   * @param {{}} props 
+   */
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isFetching: true,
+    };
+  }
+
+  /**
+   * Loads user authorization data after React has initialized the component.
+   * @override
+   */
+  componentDidMount() {
+    this.fetchAuthorizationStatus();
+  }
+
+  /**
+   * Fetches the authorization status of the user and updates the button.
+   */
+  fetchAuthorizationStatus() {
+    const fetchRequest = standardizeFetchErrors(fetch('/auth'),
+        'Failed to communicate with the server, please try again later.',
+        'Encountered a server error, please try again later.');
+
+    fetchRequest.then((authData) => {
+      console.log('Auth data received:');
+      console.log(authData);
+      this.setState({
+        isFetching: false,
+        authData,
+      });
+    }).catch((errorResponse) => {
+      console.error(
+          `Error ${errorResponse.statusCode}: ${errorResponse.error}`);
+    });
+  }
+
+  /**
+   * Renders the Login/Logout button. If the authorization data is still
+   * loading, the button is disabled.
+   * @return {React.Component}
+   * @override
+   */
+  render() { 
```

Comment:
Trailing spaces seem like the nittiest nit, but I suppose if you push this branch again you may as well remove them so we can get rid of these linting messages.

Reply:
Yes please. Let's get rid of all linter errors before merging anything to master :)

Reply:
Sgtm! Thanks for the reminder. Will update those now.